.github/workflows/ci.yml

```diff
@@ -50,6 +50,15 @@ jobs:
           access_token: ${{ secrets.GITHUB_TOKEN }}
       - name: Checkout Repository
         uses: actions/checkout@v2
+      - name: Cache Go controller tools
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-tools-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-tools-
```

Comment:
what is the reason for configuring `restore-keys`?

Reply:
it's used to search for partial matches if there's a cache miss for the key.

I just used the example from here btw https://github.com/actions/cache/blob/main/examples.md#go---modules